enginetest/queries/query_plans.go

```diff
@@ -316,11 +373,9 @@ var PlanTests = []QueryPlanTest{
 		Query: `with cte(a,b) as (select * from ab) select * from xy where exists (select * from cte where a = x)`,
 		ExpectedPlan: "SemiJoin(cte.a = xy.x)\n" +
 			" ├─ Table(xy)\n" +
-			" └─ HashLookup(child: (cte.a), lookup: (xy.x))\n" +
```

Comment:
this was an invalid plan, partial joins are not correct with hash lookups yet. need a new operator to support this.